tests: userspace: improve code coverage for userspace.c

LixinGuoX opened the pull request:
This PR includes 2 commits:
1 is for testing invalid kernel objects
2 is for supporting K_OBJ_CONDVAR in test_alloc_kobjects()

This improves code coverage for kernel/userspace.c

Signed-off-by: Lixin Guo <lixinx.guo@intel.com>

enjiamai commented on tests/kernel/mem_protect/mem_protect/src/kobject.c:
```diff
@@ -352,18 +352,19 @@ void test_kobject_release_from_user(void)
-/* @brief Test release kernel a invaild kobject
+/* @brief Test release and access grant an invaild kobject
  *
  * @details Validate release kernel objects with NULL parameter.
```
this can be also updated.

enjiamai commented on tests/kernel/mem_protect/mem_protect/src/kobject.c:
```diff
@@ -352,18 +352,19 @@ void test_kobject_release_from_user(void)
  * @ingroup kernel_memprotect_tests
  */
-void test_kobject_release_null(void)
+void test_kobject_null(void)
```
test_kobject_invalid() might be more suitable.

enjiamai commented on tests/kernel/mem_protect/userspace/src/main.c:
```diff
@@ -829,6 +829,12 @@ void test_object_recycle(void)
+	/* Validate recycle objects with NULL parameter */
+	ko = z_object_find(&dummy);
+	zassert_true(ko == NULL, "not an invalid object");
+	z_object_recycle(&dummy);
```
Suggests we add a perms_count check here.

LixinGuoX replied:
@enjiamai , since `ko` here is NULL, so we can not check  perms_count here.

enjiamai reviewed (changes requested):
could you please also help to add the @detial and @brief for test_object_recycle, thanks!

Commit:
tests: userspace: add support for K_OBJ_CONDVAR

add support for K_OBJ_CONDVAR in test_alloc_kobjects()
This improves the code coverage.

Signed-off-by: Lixin Guo <lixinx.guo@intel.com>

Commit:
tests: userspace: add support for validating invalid objects

test userspace API with invalid kernel objects.
This improves code coverage.

Signed-off-by: Lixin Guo <lixinx.guo@intel.com>

LixinGuoX commented:
Hi @enjiamai , I have updated in latest changes, could you help to review this PR again.

enjiamai reviewed (changes requested):
Please address the build failure due to number of thread exceeded.
